### ISSUE TITLE
Synchronizing calendar gets stuck

### ISSUE THREAD

**Issue by glen-84** [kewisch/gdata-provider#818]
### Discussed in https://github.com/kewisch/gdata-provider/discussions/816

<div type='discussions-op-text'>

<sup>Originally posted by **glen-84** October 11, 2024</sup>
![image](https://github.com/user-attachments/assets/c41ede1e-36ef-403d-b236-8d99f6f1afc8)

It just stays like this. Sometimes there are multiple entries in the activity manager for the same thing.

Is there some way to debug this? Any logs?

128.3.1esr (64-bit)</div>

**Comment by glen-84** [comment]
@kewisch Any ideas?

**Comment by kewisch** [comment]
How long does it stay like this? Do you get any errors in the logs? See the wiki for more info.

**Comment by glen-84** [comment]
> How long does it stay like this?

Until you restart TB.

> Do you get any errors in the logs? See the wiki for more info.

Thanks for the instructions. In terms of errors and warnings, I'm seeing this:

```
Assert failed: Calendar manager initialised calendars before loadCalendarComponent ran on the first 3pane window. This should not happen.
 calUtils.sys.mjs:131:15
    ASSERT resource:///modules/calendar/calUtils.sys.mjs:131
    loadCalendarComponent chrome://calendar/content/calendar-chrome-startup.js:41
    _loadComponentsAtStartup chrome://messenger/content/messenger.js:294
    _delayedStartup chrome://messenger/content/messenger.js:284

[calCachedCalendar] replay action failed: <unknown>, uri=googleapi://my-email@gmail.com/?calendar=%23contacts%40group.v.calendar.google.com, result=[object Object], operation=[object Object]
```

... but the 2nd error seems to be related to a different calendar (Contacts).

I also noticed an HTTP 410 on a request involving the holiday calendar:

```
Calendar: [calGoogleCalendar] Request GET https://www.googleapis.com/calendar/v3/calendars/en.sa%23holiday%40group.v.calendar.google.com/events?maxResults=1000&eventTypes=default&eventTypes=focusTime&eventTypes=outOfOffice&showDeleted=true&syncToken=*** responded with HTTP 410 gdataRequest.jsm:402:9
Calendar: [calGoogleCalendar] Server did not accept incremental update, resetting calendar and starting over. gdataCalendar.jsm:909:15
Calendar: [calGoogleCalendar] Resetting last updated counter for Holidays in South Africa gdataCalendar.jsm:742:11
```

Although later on it seems to go through ...

```
Calendar: [calGoogleCalendar] Request GET https://www.googleapis.com/calendar/v3/calendars/en.sa%23holiday%40group.v.calendar.google.com/events?maxResults=1000&eventTypes=default&eventTypes=focusTime&eventTypes=outOfOffice responded with HTTP 200 gdataRequest.jsm:402:9
Calendar: [calGoogleCalendar] Timezone for Holidays in South Africa is Africa/Johannesburg gdataUtils.jsm:1114:11
Calendar: [calGoogleCalendar] Parsing 250 received events
```

The first request includes two additional query parameters (`showDeleted=true`  and `syncToken=...`).

**Comment by kewisch** [comment]
Yes the 410 causes a full sync. If the calendar is large, then the sync will take super long for reasons beyond my control. I've heard from folks where it took like 30 minutes of freezing and then unfroze. 

Does the 410 happen every time you restart? Does re-subscribing to the calendar help at all?

**Comment by glen-84** [comment]
> Does the 410 happen every time you restart?

Yes.

> Does re-subscribing to the calendar help at all?

I unsubscribed via Thunderbird and via Google Calendar, and then re-subscribed via both, and the calendar was added using CalDAV. I noticed that the identifier now starts with `en-gb.sa`, while it was `en.sa` before. I'm not sure if that locale difference matters.

Anyway, I think that it may be fixed now. :tada:

Thanks a lot for your help!
